pretyping/pretyping.ml

```diff
@@ -487,7 +487,15 @@ let rec pretype k0 resolve_tc (tycon : type_constraint) (env : GlobEnv.t) (sigma
       let evk =
         try Evd.evar_key id sigma
         with Not_found ->
-          user_err ?loc  (str "Unknown existential variable.") in
+          let all_undef_evars = Evar.Set.elements (Evar.Map.domain (Evd.undefined_map sigma)) in
+          if List.mem id (List.map (fun ev -> evar_suggested_name ev sigma) all_undef_evars) then
```

Comment:
This is going to be horribly slow for a mere debugging feature.

Reply:
To be on the safe side, I delayed the printing of the message to `himsg.ml` and made the test of belonging to the list of evar lazy (so that adding a number suffix does not systematically induce a n^2 complexity in case all evars have the same name, say `Goal`). But an LtacX funciton is (in principle) not supposed to use a `?x` with an `x` not bound by the LtacX function. So, the situations where the error would be raised and would not go directly to the toplevel seem rare (e.g a complex tactic expression with backtracking in strict mode?).

(Incidentally, I think there is a misunderstanding. It is not about debugging but about improving a user-level message.)

But you are right, let us be on the safe side. As far as I can judge, the worse we can have is `n` evars with same name `id` and asking for `?idN` for `N`<`n`, which shall be in complexity `N`^2.